src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service;
+
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Map<String, Integer> repository;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        repository = new HashMap<>();
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        fruitTransactions.stream()
+                .forEach(new Consumer<FruitTransaction>() {
+                    @Override
+                    public void accept(FruitTransaction fruitTransaction) {
+                        String fruitkey = fruitTransaction.getFruit();
+                        OperationHandler operationHandler = operationStrategy
+                                .getOperationHandler(fruitTransaction.getOperation())
+                                .getValue();
```

Comment:
The call to 'getValue()' is suspicious because 'getOperationHandler()' should already return an 'OperationHandler'. Please ensure that you are using the correct method from the 'OperationStrategy' interface.